src/core/stream/representation/representation_stream.ts

```diff
@@ -209,8 +210,21 @@ export default function RepresentationStream<TSegmentDataType>(
     segmentsLoadingCanceller.signal,
   );
 
+  const canStream = new SharedReference<boolean | undefined>(undefined);
```

Comment:
I'm unsure of why ever setting it to `undefined`.
Why not initially `true` or `false`?